clippy_lints/src/non_octal_unix_permissions.rs

```diff
@@ -51,26 +76,22 @@ impl<'tcx> LateLintPass<'tcx> for NonOctalUnixPermissions {
                         ))
                         || (path.ident.name == sym!(set_mode)
                             && cx.tcx.is_diagnostic_item(sym::FsPermissions, adt.did())))
-                    && let ExprKind::Lit(_) = param.kind
+                    && let ExprKind::Lit(lit_kind) = param.kind
                     && param.span.eq_ctxt(expr.span)
+                    && let Some(snip) = snippet_opt(cx, param.span)
+                    && !(snip.starts_with("0o") || check_binary_unix_permissions(&lit_kind.node, &snip))
```

Comment:
I'd say ignoring `.starts_with("0b")` is sufficient, weirdly formatted literals could happen but it's not very likely

Reply:
fine, I'll fix this.